### ISSUE TITLE
Navigation "lost" when going from talk to planning

### ISSUE THREAD

**Issue by xballoy** [MielPoPSCrew/DevoxxFR-2018-PWA#7]
Hello,

Great job! Thanks to you I will have my Devoxx planning in my pocket next week :)

However when I was browsing from my mobile I realised something kind of anoying:

1. Go to planning
2. Switch the day (do not stay on _mercredi_)
3. Click on any talk
4. Press back 
5. You are lost: not on the day you selected, the scrolls (horizontal and vertical have changed...)

**Comment by clementgarin** [comment]
Hi, thanks for your feedback !

Concerning the scrolls, the issue is global to the application. I will try to use the solution from https://github.com/angular/angular/issues/10929#issuecomment-372886603 

The tabs now... One possible solution is to use the route system : for example, `/schedule/friday` to keep a trace of the last selected tab in the browser history. But if you play to much with tabs, you will have to press 'back' serveral times before reaching another app section :/

Any opinions ?

**Comment by xballoy** [comment]
I agree with you, if everytime we change tab we add something in the browser history we will have a strange behaviour.

However, why not pushing the current schedule route in the history only when the user navigate to the detail?

**Comment by clementgarin** [comment]
In fact, pushing the current route in the history is handled by Angular itself, and I don't think it's possible to change the behavior.

I will test this : 

- Creating a service to store navigation states (like schedule tab index)
- Adding a subscription to the Angular Location service in the Schedule component to track if the back button was pressed
- Retrieved the tab index from the navigation state service

**Comment by clementgarin** [comment]
These problems are now partially solved:

The tab index is saved when navigating back/next in bookmark/schedule.

Concerning the scroll restoration, it works... Only on mobile view (when navigation menu is hidden). I don't know why, the scroll position isn't restored on desktop view. Maybe something relative to media-query ? (cc @leodonny ).

You can test this behavior on the [bugfix/#7-navigation_lost  branch](https://github.com/MielPoPSCrew/DevoxxFR-2018-PWA/tree/bugfix/%237-navigation_lost)

Should we consider embedded these corrections into the next release, even it's not working on desktop view ?

**Comment by xballoy** [comment]
Maybe we can split the issue in two:
- the navigation issue: ✅ 
- the scroll issue

If the navigation issue is solved we can ship it IMO
